app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -602,20 +648,34 @@ class HomeFragment : Fragment() {
         )
     }
 
+    private fun updateSearchSelectorMenu(searchEngines: List<SearchEngine>) {
+        val searchEngineList = searchEngines
+            .map {
+                TextMenuCandidate(
+                    text = it.name,
+                    start = DrawableMenuIcon(
+                        drawable = it.icon.toDrawable(resources),
+                    ),
+                ) {
+                    sessionControlInteractor.onMenuItemTapped(SearchSelectorMenu.Item.SearchEngine(it))
+                }
+            }
+
+        searchSelectorMenu.menuController.submitList(searchSelectorMenu.menuItems(searchEngineList))
+    }
+
     private fun observeSearchEngineChanges() {
         consumeFlow(store) { flow ->
             flow.map { state -> state.search.selectedOrDefaultSearchEngine }
                 .ifChanged()
                 .collect { searchEngine ->
-                    if (searchEngine != null) {
-                        val iconSize =
-                            requireContext().resources.getDimensionPixelSize(R.dimen.preference_icon_drawable_size)
-                        val searchIcon =
-                            BitmapDrawable(requireContext().resources, searchEngine.icon)
-                        searchIcon.setBounds(0, 0, iconSize, iconSize)
-                        binding.searchEngineIcon.setImageDrawable(searchIcon)
```

Comment:
Trying to understand if this change was intentional. I see we aren't setting the bounds anymore, but I am wondering what is the reasoning behind the change? I see we do call this type `setBounds(0, 0, iconSize, iconSize)` in many places in our codebase https://searchfox.org/mozilla-mobile/search?q=setBounds&path=, but I don't have any context on its significance or usage. Since we are removing it here, I would like to learn a bit more what types of problem we were encountering and solving.

Reply:
yes, I was talking about it [here](https://github.com/mozilla-mobile/fenix/pull/27647#issuecomment-1301656042).

short version, this particular code doesn't do anything, since we create a `BitmapDrawable` from a `Bitmap`. Changing it's bounds afterwards doesn't affect the Bitmap itself, so it's height and width will be used.
You can see that by changing `preference_icon_drawable_size` to 2 or 200 – it won't have any effect.